contracts/citycoin.clar

```diff
@@ -178,9 +179,35 @@
 )
 
 ;; Function for deciding how many tokens to mint, depending on when they were mined.
-;; Tailor to your own needs.
-(define-read-only (get-coinbase-amount (stacks-block-ht uint))
-    u500000000
+(define-read-only (get-coinbase-amount (miner-block-height uint))
+    (let
+        (
+            ;; set a new variable to make things easier to read
+            (activation-block-height (var-get first-stacking-block))
+        )
+
+        ;; determine if mining was active, return 0 if not
+        (asserts! (>= miner-block-height activation-block-height) u0)
+
+        ;; evaluate current block height against issuance schedule and return correct coinbase amount
+        ;; halvings occur every 210,000 blocks for 1,050,000 Stacks blocks
+        ;; then mining continues indefinitely with 3,125 CityCoins as the reward
+
+        (asserts! (> miner-block-height (+ activation-block-height MINING-HALVING-BLOCKS))
+            (if (<= (- miner-block-height activation-block-height) u10000)
+                u250000 ;; bonus reward first 10,000 blocks
+                u100000 ;; standard reward remaining 200,000 blocks until 1st halving
+            )
+        )
+        (asserts! (> miner-block-height (+ activation-block-height (* u2 MINING-HALVING-BLOCKS))) u50000)        ;; between 1st and 2nd halving u50000
+        (asserts! (> miner-block-height (+ activation-block-height (* u3 MINING-HALVING-BLOCKS))) u25000) ;; between 2nd and 3rd halving u25000
+        (asserts! (> miner-block-height (+ activation-block-height (* u4 MINING-HALVING-BLOCKS))) u12500) ;; between 3rd and 4th halving u12500
+        (asserts! (> miner-block-height (+ activation-block-height (* u5 MINING-HALVING-BLOCKS))) u6250)  ;; between 4th and 5th halving u6250
+
+        ;; default value after 5th halving
+        u3125
```

Comment:
```suggestion
        (asserts! (> miner-block-height (+ activation-block-height MINING-HALVING-BLOCKS))
            (if (<= (- miner-block-height activation-block-height) u10000)
                u250000000000 ;; bonus reward first 10,000 blocks
                u100000000000 ;; standard reward remaining 200,000 blocks until 1st halving
            )
        )
        (asserts! (> miner-block-height (+ activation-block-height (* u2 MINING-HALVING-BLOCKS))) u50000000000)        ;; between 1st and 2nd halving u50000
        (asserts! (> miner-block-height (+ activation-block-height (* u3 MINING-HALVING-BLOCKS))) u25000000000) ;; between 2nd and 3rd halving u25000
        (asserts! (> miner-block-height (+ activation-block-height (* u4 MINING-HALVING-BLOCKS))) u12500000000) ;; between 3rd and 4th halving u12500
        (asserts! (> miner-block-height (+ activation-block-height (* u5 MINING-HALVING-BLOCKS))) u6250000000)  ;; between 4th and 5th halving u6250

        ;; default value after 5th halving
        u3125000000
```

Reply:
Do you mean from `u6` to `u0`? That was the change I made in https://github.com/citycoins/citycoin/pull/35/commits/a0efee5ce7f1ebd3f33cb05edc1f0b68fc8e2074 to avoid dealing with micro-CityCoins and the additional zeroes in the math. 

A CityCoin is a CityCoin, so unless there is a need for a decimal amount, it's easier to just let it be a whole number. Plus, imagining this going live for several cities in the future, trying to designate the difference between a $MIA and $uMIA is only going to cause more confusion.

I'm a fan of simple.

Reply:
Please forget about it. 
Looks like I was tired and sleepy to make good code review.
Somehow I saw that you changed from `u0` to `u6` while in reality you did the opposite. An of course this change was 100% correct and my suggestion are bad and should be discarded.